CHANGELOG.md

```diff
@@ -1,5 +1,10 @@
 # Unreleased
 
+## Expression Functions
+- Added `nonNull` for special evaluation handling for temporary null values. This may be useful in migrating code that is now producing many incompatbile argument warnings.
```

Comment:
I think it would be useful to explain a bit more about what `nonNull` does here. I assume it blocks the input until it's non-null, right?

Reply:
I've provided a few more details on the API docs.  Yes, it "blocks" input. We really need to agree on a term of what "lostData" means.  It's not quite blocking, since existing data will be cleared when lost.

Maybe once i'm done some more refactoring I'll write up a clearer doc on expressions and some null handling.

Reply:
Sounds good. We can be a bit vague in the mean time.